src/stream/stream/mod.rs

```diff
@@ -484,7 +486,12 @@ extension_trait! {
 
             let s: VecDeque<&str> = vec!["1", "lol", "3", "NaN", "5"].into_iter().collect();
 
-            let mut parsed = s.filter_map(|a| a.parse::<u32>().ok());
+            let mut parsed = s.filter_map(|a|
+                async move {
+                    a.parse::<u32>().ok()
+                });
+
+            let mut parsed = unsafe { std::pin::Pin::new_unchecked(&mut parsed) };
```

Comment:
Hm is there something we could do about the `unsafe` block here? Even using `pin_utils` might be better than what we currently have, but I'm curious why we need it?

Reply:
yep, that's must be some leftover